primitives/src/trie/mpt.rs

```diff
@@ -71,6 +71,15 @@ pub enum Error {
     LegacyRlp(#[from] DecoderError),
 }
 
+impl From<B256> for MptNode {
+    fn from(value: B256) -> Self {
+        MptNode {
+            data: MptNodeData::Digest(value),
```

Comment:
For the EMPTY_ROOT this should be Null instead.

There is already `node_from_digest` which does that, so we should consider consolidating.

However, it is important to note that sometimes the API returns B256::ZERO when it should instead be EMPTY_ROOT. So to be on the safe side we should:
```rust
    match digest {
        EMPTY_ROOT | B256::ZERO => MptNode::default(),
        _ => MptNodeData::Digest(digest).into(),
    }
```